subsys/settings/CMakeLists.txt

```
@@ -5,7 +5,7 @@ zephyr_include_directories(
   include
   )
 
-if (CONFIG_SETTINGS_FCB OR CONFIG_SETTINGS_NVS)
+if (ZEPHYR_NRF_MODULE_DIR AND (CONFIG_SETTINGS_FCB OR CONFIG_SETTINGS_NVS))
```

Same reason as above:
```suggestion
if((COMMAND add_partition_manager_config) AND (CONFIG_SETTINGS_FCB OR CONFIG_SETTINGS_NVS))
```

---

If you consider #279 (comment) and the conclusion is that we don't need the check anymore, would this comment still be valid?

---

Based on your explanation, I think it is still important to safeguard the call to `add_partition_manager_config`, but a personal preference would be to test that the function exist (as suggested above) instead of testing for nRF module.

In principle code doesn't care if `add_partition_manager_config` is defined in `zephyr/cmake/extensions.cmake`, `nrf/cmake/extensions.cmake`, or a 3rd place.
It only requires this function to be defined.

But will not block PR on this, as both solutions are acceptable.